org.eclipse.ice.modeling/src/main/java/org/eclipse/ice/modeling/workflowEngine/Procedure.java

```diff
@@ -0,0 +1,88 @@
+package org.eclipse.ice.modeling.workflowEngine;
+
+/**
+ * The Procedure hold the set of steps that constitute the workflow.  
+ * My current thinking is that this class does not hold state, only Steps
+ * and Completion Criteria
+ */
+public class Procedure {
+
+	private Step[] steps;    // This should really be a List where steps can be added dynamically
```

Comment:
I agree that this should be a list, we may also want to consider making it a hashmap if there are reasons to be able to identify individual steps quickly via a string identifier or something like this.

Reply:
Agree, here again my lack of familiarity and comfort with Java is limiting.  I did this just to prove concepts.  You will see this in multiple places.  I too was thinking hash map, but from what little I read in Java, List is an interface that works with hash maps and other types of Collections, etc.  Just don't know how to work with them yet.